### PR TITLE
helm: add ability to add custom environment variable

### PR DESCRIPTION
**Pull request by zufardhiyaulhaq**
Add ability to add custom environment variables to the container deployment.

### CODE REVIEW

**Comment by zufardhiyaulhaq** [comment]
cc @anguslees

File: helm/sealed-secrets/templates/deployment.yaml [hunk 1]
```diff
@@ -57,6 +57,12 @@ spec:
           volumeMounts:
           - mountPath: /tmp
             name: tmp
+          {{- if .Values.env }}
+          env:
+            {{- with .Values.env }}
+              {{- toYaml . | nindent 12 }}
+            {{- end }}
+          {{- end }}
```
Comment (juan131) [inline helm/sealed-secrets/templates/deployment.yaml:65]: This is my suggestion for supporting custom env vars from different sources:
```suggestion
         {{- if .Values.extraEnvVars }}
         env: {{- include "sealed-secrets.render" (dict "value" .Values.extraEnvVars "context" $) | nindent 12 }}
          {{- end }}
          {{- if or .Values.extraEnvVarsCM .Values.extraEnvVarsSecret }}
          envFrom:
            {{- if .Values.extraEnvVarsCM }}
            - configMapRef:
                name: {{ .Values.extraEnvVarsCM }}
            {{- end }}
            {{- if .Values.extraEnvVarsSecret }}
            - secretRef:
                name: {{ .Values.extraEnvVarsSecret }}
            {{- end }}
          {{- end }}
```

File: helm/sealed-secrets/values.yaml [hunk 2]
```diff
@@ -22,6 +22,11 @@ controller:
 # namespace: Namespace to deploy the controller.
 namespace: ""
 
+# Extra env variables to to be added to controller deployment
+env: []
+# - name: SEALED_SECRETS_UPDATE_STATUS
+#   value: 1
```
Comment (juan131) [inline helm/sealed-secrets/values.yaml:28]: Following the previous recommendation:
```suggestion
## @param extraEnvVars Array with extra environment variables to add to controller container
## e.g:
## extraEnvVars:
##   - name: FOO
##     value: "bar"
##
extraEnvVars: []
## @param extraEnvVarsCM Name of existing ConfigMap containing extra env vars for controller container
##
extraEnvVarsCM: ""
## @param extraEnvVarsSecret Name of existing Secret containing extra env vars for controller container
##
extraEnvVarsSecret: ""
```

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks so much for this PR @zufardhiyaulhaq !! Please check my recommendations when you have a chance.

**Comment by zufardhiyaulhaq** [comment]
@juan131 I am not sure why we only allow a single configmap and secret to be able to mount to the deployment. I think it's not a generic approach

**Comment by juan131** [comment]
What do you mean @zufardhiyaulhaq ? Are you suggesting that `extraEnvVarsCM` and `extraEnvVarsSecret` properties should be an array?

**Comment by zufardhiyaulhaq** [comment]
yes @juan131 wdyt? it's better for array right?

**Pull request closed without merging**